Sources/SemanticIndex/SemanticIndexManager.swift

```diff
@@ -488,7 +468,10 @@ public final actor SemanticIndexManager {
         // setting it to `.scheduled` because we don't have an `await` call between the creation of `indexTask` and
         // this loop, so we still have exclusive access to the `SemanticIndexManager` actor and hence `updateIndexStore`
         // can't execute until we have set all index statuses to `.scheduled`.
-        indexStatus[file.sourceFile] = .scheduled((taskID, indexTask))
+        inProgressIndexTasks[file.sourceFile] = .waitingForPreparation(
+          preparationTaskID: preparationTaskID,
+          indexTask: indexTask
+        )
```

Comment:
Could we potentially run into a case?

1. We kick off indexing for a single file
2. We then schedule the indexing for a whole bunch of files that include that file
3. `inProgressIndexTasks` gets set to the new preparation task id for the large group of files
4. The indexing job for the single file gets completed, but that won't be reflected in `inProgressIndexTasks`, so `inProgressIndexFiles` will still list it as running
5. We then have to wait for the large file job to complete before the file is marked as up-to-date

Reply:
Yes, we can run into that case but I don’t think it matters because the two uses of `inProgressIndexFiles` are:
- Index progress reporting: The effect of this will be that the numerator of the progress will only be incremented to reflect that the single file is up-to-date only when the batch is indexed, ie. it’s lower by 1 for a little while longer. Since we’re also waiting for other files to be indexed, I don’t think it actually matters
- Waiting for an up-to-date index: In that case we want to wait for all index tasks to finish anyway, including the batch.